torch/lib/c10d/ProcessGroup.cpp

```diff
@@ -145,31 +146,25 @@ void ProcessGroup::Work::abort() {
 }
 
 c10::intrusive_ptr<c10::ivalue::Future> ProcessGroup::Work::getFuture() {
-  TORCH_CHECK(false, "ProcessGroup::Work::getFuture not implemented.")
+  return future_;
+}
+
+void ProcessGroup::Work::finish(c10::IValue value) {
```

Comment:
Since we always expect this value to be a list of tensors, could we maybe use a more specialized type so that we can catch any mismatch more clearly?

Reply:
This value can also be a list of list of tensors in some cases of ops that GLOO supports